mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td

```diff
@@ -40,26 +40,27 @@ def Mesh_ClusterOp : Mesh_Op<"cluster", [Symbol]> {
     determine the layout and the addressing space of the computation distributed
     across the mesh.
 
-    3. `dim_sizes`: This attribute represents the device assignment along the
-    axes of the cluster. Each integer in the array corresponds to the number of
-    devices along a specific axis. If an integer value is 0, it implies that the
-    number of devices along that axis is unknown. This flexibility allows for
-    dynamic device assignment or configurations where the exact number of
-    devices might not be determined during compile time.
+    3. `dim_sizes`: This attribute represents the shape of the device cluster.
+    It uses the same notation as a tensor shape. Also allowing for dynamic
```

Comment:
The tensor shape doesn't have the `rank` attribute. So do we need to remove rank and get it from `dim_sizes.size()` instead?

Reply:
I think we can remove the `rank` attribute. I don't envision a scenario where we would have to describe a cluster shape with many trailing dynamic dimensions like
```
2x3x?x?x?x?x?x?x?x?
```
Famous last words.

Should I remove it in this PR?

Reply:
If it's not about the formatting, I rather keep changes separate if possible.